functions/Get-DbaDbSchema.ps1

```diff
@@ -0,0 +1,126 @@
+
+    .PARAMETER WhatIf
+        Shows what would happen if the command were to run. No actions are actually performed.
+
+    .PARAMETER Confirm
+        Prompts you for confirmation before executing any changing operations within the command.
+
+    .PARAMETER EnableException
+        By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+        This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+        Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+    .NOTES
+        Tags: Data, Database, Migration, Permission, Security, Schema, Table, User
+        Author: Adam Lancaster https://github.com/lancasteradam
+
+        dbatools PowerShell module (https://dbatools.io)
+        Copyright: (c) 2021 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+        https://dbatools.io/Get-DbaDbSchema
+
+    .EXAMPLE
+        PS C:\> Get-DbaDbSchema -SqlInstance localhost
+
+        Gets all non-system database schemas from all user databases on the localhost instance. Note: the dbo schema is a system schema and won't be included in the output from this example. To include the dbo schema specify -IncludeSystemSchemas
+
+    .EXAMPLE
+        PS C:\> Get-DbaDbSchema -SqlInstance localhost -SchemaName dbo -IncludeSystemSchemas
+
+        Returns the dbo schema from the databases on the localhost instance.
+
+    .EXAMPLE
+        PS C:\> Get-DbaDbSchema -SqlInstance localhost -IncludeSystemDatabases -IncludeSystemSchemas
+
+        Gets all database schemas from all databases on the localhost instance.
+
+    .EXAMPLE
+        PS C:\> Get-DbaDbSchema -SqlInstance localhost -SchemaName TestSchema
+
+        Finds and returns the TestSchema schema from the localhost instance.
+
+    .EXAMPLE
+        PS C:\> Get-DbaDbSchema -SqlInstance localhost -SchemaOwner DBUser1
+
+        Finds and returns the schemas owned by DBUser1 from the localhost instance.
+
+    .EXAMPLE
+        PS C:\> Get-DbaDbSchema -SqlInstance localhost -Database TestDB -SchemaOwner DBUser1
+
+        Finds and returns the schemas owned by DBUser1 in the TestDB database from the localhost instance.
+
+    .EXAMPLE
+        PS C:\> $schema = Get-DbaDbSchema -SqlInstance localhost -Database TestDB -SchemaName TestSchema
+        PS C:\> $schema.Owner = DBUser2
+        PS C:\> $schema.Alter()
+
+        Finds the TestSchema in the TestDB on the localhost instance and then changes the schema owner to DBUser2
+
+    .EXAMPLE
+        PS C:\> $schema = Get-DbaDbSchema -SqlInstance localhost -Database TestDB -SchemaName TestSchema
+        PS C:\> $schema.Drop()
+
+        Finds the TestSchema in the TestDB on the localhost instance and then drops it. Note: to drop a schema all objects must be transferred to another schema or dropped.
```

Comment:
Can we add an example that demonstrates piping into this function?

Reply:
yes, will do